coap/api/transport.go

```diff
@@ -66,75 +67,87 @@ func MakeCoAPHandler(svc coap.Service, l *slog.Logger) mux.HandlerFunc {
 	return handler
 }
 
-func sendResp(w mux.ResponseWriter, resp *message.Message) {
-	if err := w.Client().WriteMessage(resp); err != nil {
+func sendResp(w mux.ResponseWriter, resp *pool.Message) {
+	if resp.Code() == codes.Content {
+		obs, err := resp.Options().Observe()
+		if err != nil {
+			logger.Warn(fmt.Sprintf("Error reading observe option: %s", err))
+		}
+		if obs == startObserve {
+			return
+		}
+	}
```

Comment:
@1998-felix This was handled already in `handleGet`, Do we need this again here ?

Reply:
It is there so that a nil packet response will not be sent on observe.